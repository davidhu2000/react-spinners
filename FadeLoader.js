(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports', 'emotion', 'react', 'prop-types', '@emotion/core', 'recompose/onlyUpdateForKeys'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require('emotion'), require('react'), require('prop-types'), require('@emotion/core'), require('recompose/onlyUpdateForKeys'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.emotion, global.react, global.propTypes, global.core, global.onlyUpdateForKeys);
    global.FadeLoader = mod.exports;
  }
})(this, function (exports, _emotion, _react, _propTypes, _core, _onlyUpdateForKeys) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });

  var _react2 = _interopRequireDefault(_react);

  var _propTypes2 = _interopRequireDefault(_propTypes);

  var _onlyUpdateForKeys2 = _interopRequireDefault(_onlyUpdateForKeys);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  function _classCallCheck(instance, Constructor) {
    if (!(instance instanceof Constructor)) {
      throw new TypeError("Cannot call a class as a function");
    }
  }

  var _createClass = function () {
    function defineProperties(target, props) {
      for (var i = 0; i < props.length; i++) {
        var descriptor = props[i];
        descriptor.enumerable = descriptor.enumerable || false;
        descriptor.configurable = true;
        if ("value" in descriptor) descriptor.writable = true;
        Object.defineProperty(target, descriptor.key, descriptor);
      }
    }

    return function (Constructor, protoProps, staticProps) {
      if (protoProps) defineProperties(Constructor.prototype, protoProps);
      if (staticProps) defineProperties(Constructor, staticProps);
      return Constructor;
    };
  }();

  function _possibleConstructorReturn(self, call) {
    if (!self) {
      throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
    }

    return call && (typeof call === "object" || typeof call === "function") ? call : self;
  }

  function _inherits(subClass, superClass) {
    if (typeof superClass !== "function" && superClass !== null) {
      throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
    }

    subClass.prototype = Object.create(superClass && superClass.prototype, {
      constructor: {
        value: subClass,
        enumerable: false,
        writable: true,
        configurable: true
      }
    });
    if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
  }

  var fade = (0, _core.keyframes)('50%{opacity:0.3}100%{opacity:1}');

  var Loader = function (_React$Component) {
    _inherits(Loader, _React$Component);

    function Loader() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, Loader);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Loader.__proto__ || Object.getPrototypeOf(Loader)).call.apply(_ref, [this].concat(args))), _this), _this.style = function (i) {
        var _this$props = _this.props,
            height = _this$props.height,
            width = _this$props.width,
            margin = _this$props.margin,
            color = _this$props.color,
            radius = _this$props.radius,
            widthUnit = _this$props.widthUnit,
            heightUnit = _this$props.heightUnit,
            radiusUnit = _this$props.radiusUnit;


        return (0, _core.css)('{position:absolute;width:', '' + width + widthUnit, ';height:', '' + height + heightUnit, ';margin:', margin, ';background-color:', color, ';border-radius:', '' + radius + radiusUnit, ';transition:2s;animation-fill-mode:\'both\';animation:', fade, ' 1.2s ', i * 0.12, 's infinite ease-in-out;}');
      }, _this.radius = 20, _this.quarter = _this.radius / 2 + _this.radius / 5.5, _this.wrapper = function () {
        var wrapper = (0, _core.css)('{position:relative;font-size:0;top:', _this.radius, 'px;left:', _this.radius, 'px;width:', _this.radius * 3, 'px;height:', _this.radius * 3, 'px;}');

        return _this.props.css ? (0, _core.css)(wrapper, ';', _this.props.css) : wrapper;
      }, _this.a = function () {
        return (0, _core.css)(_this.style(1), ';top:', _this.radius, 'px;left:0;');
      }, _this.b = function () {
        return (0, _core.css)(_this.style(2), ';top:', _this.quarter, 'px;left:', _this.quarter, 'px;transform:rotate(-45deg);');
      }, _this.c = function () {
        return (0, _core.css)(_this.style(3), ';top:0;left:', _this.radius, 'px;transform:rotate(90deg);');
      }, _this.d = function () {
        return (0, _core.css)(_this.style(4), ';top:', -_this.quarter, 'px;left:', _this.quarter, 'px;transform:rotate(45deg);');
      }, _this.e = function () {
        return (0, _core.css)(_this.style(5), ';top:', -_this.radius, 'px;left:0;');
      }, _this.f = function () {
        return (0, _core.css)(_this.style(6), ';top:', -_this.quarter, 'px;left:', -_this.quarter, 'px;transform:rotate(-45deg);');
      }, _this.g = function () {
        return (0, _core.css)(_this.style(7), ';top:0;left:', -_this.radius, 'px;transform:rotate(90deg);');
      }, _this.h = function () {
        return (0, _core.css)(_this.style(8), ';top:', _this.quarter, 'px;left:', -_this.quarter, 'px;transform:rotate(45deg);');
      }, _temp), _possibleConstructorReturn(_this, _ret);
    }

    _createClass(Loader, [{
      key: 'render',
      value: function render() {
        var loading = this.props.loading;


        return loading ? _react2.default.createElement(
          'div',
          {
            className: (0, _emotion.css)(this.wrapper())
          },
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.a())
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.b())
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.c())
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.d())
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.e())
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.f())
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.g())
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.h())
          })
        ) : null;
      }
    }]);

    return Loader;
  }(_react2.default.Component);

  Loader.propTypes = {
    loading: _propTypes2.default.bool,
    color: _propTypes2.default.string,
    height: _propTypes2.default.number,
    width: _propTypes2.default.number,
    margin: _propTypes2.default.string,
    radius: _propTypes2.default.number,
    heightUnit: _propTypes2.default.string,
    widthUnit: _propTypes2.default.string,
    radiusUnit: _propTypes2.default.string,
    css: _propTypes2.default.string
  };

  Loader.defaultProps = {
    loading: true,
    color: '#000000',
    height: 15,
    width: 5,
    margin: '2px',
    radius: 2,
    widthUnit: 'px',
    heightUnit: 'px',
    radiusUnit: 'px',
    css: ''
  };

  var Component = (0, _onlyUpdateForKeys2.default)(['loading', 'color', 'height', 'width', 'margin', 'radius', 'widthUnit', 'heightUnit', 'radiusUnit', 'css'])(Loader);
  Component.defaultProps = Loader.defaultProps;
  exports.default = Component;
});