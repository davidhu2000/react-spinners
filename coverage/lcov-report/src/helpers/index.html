
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/helpers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/helpers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>86/86</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.88% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>32/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>85/85</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="animation.ts"><a href="animation.ts.html">animation.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file high" data-value="colors.ts"><a href="colors.ts.html">colors.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="47" class="abs high">47/47</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="46" class="abs high">46/46</td>
	</tr>

<tr>
	<td class="file high" data-value="unitConverter.ts"><a href="unitConverter.ts.html">unitConverter.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Jun 25 2024 21:55:12 GMT-0700 (Pacific Daylight Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    