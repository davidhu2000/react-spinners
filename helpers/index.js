(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.index = mod.exports;
  }
})(this, function (exports) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  var calculateRgba = exports.calculateRgba = function calculateRgba(color, opacity) {
    if (color[0] === '#') {
      color = color.slice(1);
    }

    if (color.length === 3) {
      var res = '';
      color.split('').forEach(function (c) {
        res += c;
        res += c;
      });
      color = res;
    }

    var rgbValues = color.match(/.{2}/g).map(function (hex) {
      return parseInt(hex, 16);
    }).join(', ');
    return 'rgba(' + rgbValues + ', ' + opacity + ')';
  };
});