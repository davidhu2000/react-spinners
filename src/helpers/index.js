export const calculateRgba = (color, opacity) => {
  if (color[0] === '#') {
    color = color.slice(1);
  }

  if (color.length === 3) {
    let res = '';
    color.split('').forEach(c => {
      res += c;
      res += c;
    });
    color = res;
  }

  let rgbValues = color.match(/.{2}/g).map(hex => parseInt(hex, 16)).join(', ');
  return `rgba(${rgbValues}, ${opacity})`;
};

export * from './proptypes';
