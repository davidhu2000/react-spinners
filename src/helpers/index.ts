export * from "./proptypes";
export * from "./colors";
export * from "./unitConverter";
