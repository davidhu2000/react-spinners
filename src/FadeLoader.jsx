/** @jsx jsx */
import React from 'react';
import { keyframes, css, jsx } from '@emotion/core';
import onlyUpdateForKeys from 'recompose/onlyUpdateForKeys';
import { heightWidthRadiusKeys, heightWidthRadiusDefaults, heightWidthRadiusProps } from './helpers';

const fade = keyframes`
  50% {opacity: 0.3} 
  100% {opacity: 1}
`;

class Loader extends React.Component {
    style = i => {
      const {
        height, width, margin, color, radius, widthUnit, heightUnit, radiusUnit
      } = this.props;

      return css`{
            position: absolute;
            width: ${`${width}${widthUnit}`};
            height: ${`${height}${heightUnit}`};            
            margin: ${margin};
            background-color: ${color};
            border-radius: ${`${radius}${radiusUnit}`};
            transition: 2s;
            animation-fill-mode: 'both';
            animation: ${fade} 1.2s ${i * 0.12}s infinite ease-in-out;
        }`;
    };

  radius = 20;
  quarter = (this.radius / 2) + (this.radius / 5.5);

  wrapper = () => {
    const wrapper = css`{        
            position: relative;
            font-size: 0;
            top: ${this.radius}px;
            left: ${this.radius}px;
            width: ${this.radius * 3}px;
            height: ${this.radius * 3}px;
        }`;

    return this.props.css ? css`${wrapper};${this.props.css}` : wrapper;
  };

  a = () => css`
          ${this.style(1)};
          top: ${this.radius}px;
          left: 0;
        `;
  b = () => css`
          ${this.style(2)};
          top: ${this.quarter}px;
          left: ${this.quarter}px;
          transform: rotate(-45deg);
        `;
  c = () => css`
          ${this.style(3)};
          top: 0;
          left: ${this.radius}px;
          transform: rotate(90deg);
        `;
  d = () => css`
          ${this.style(4)};
          top: ${-this.quarter}px;
          left: ${this.quarter}px;
          transform: rotate(45deg);
        `;
  e = () => css`
          ${this.style(5)};
          top: ${-this.radius}px;
          left: 0;
        `;
  f = () => css`
          ${this.style(6)};
          top: ${-this.quarter}px;
          left: ${-this.quarter}px;
          transform: rotate(-45deg);
        `;
  g = () => css`
          ${this.style(7)};
          top: 0;
          left: ${-this.radius}px;
          transform: rotate(90deg);
        `;
  h = () => css`
          ${this.style(8)};
          top: ${this.quarter}px;
          left: ${-this.quarter}px;
          transform: rotate(45deg);
        `;

  render() {
    const { loading } = this.props;

    return loading ?
      <div css={this.wrapper()}>
        <div css={this.a()} />
        <div css={this.b()} />
        <div css={this.c()} />
        <div css={this.d()} />
        <div css={this.e()} />
        <div css={this.f()} />
        <div css={this.g()} />
        <div css={this.h()} />
      </div> : null;
  }
}

Loader.propTypes = heightWidthRadiusProps;

Loader.defaultProps = heightWidthRadiusDefaults(15, 5, 2);

const Component = onlyUpdateForKeys(heightWidthRadiusKeys)(Loader);
Component.defaultProps = Loader.defaultProps;
export default Component;
