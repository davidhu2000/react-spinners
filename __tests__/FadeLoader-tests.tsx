import * as React from "react";
import { mount, ReactWrapper } from "enzyme";
import { matchers } from '@emotion/jest';
expect.extend(matchers);

import FadeLoader from "../src/FadeLoader";
import { LoaderHeightWidthRadiusProps } from "../src/interfaces";
import { heightWidthRadiusDefaults } from "../src/helpers";

describe("FadeLoader", () => {
  let loader: ReactWrapper<LoaderHeightWidthRadiusProps, null, FadeLoader>;
  let props: LoaderHeightWidthRadiusProps;
  const defaultColor = "#000000";
  const defaultHeight = 15;
  const defaultWidth = 5;
  const defaultRadius = 2;
  const defaultMargin = 2;
  const defaultUnit = "px";

  it("should match snapshot", () => {
    loader = mount(<FadeLoader />);
    expect(loader).toMatchSnapshot();
  });

  it("should contain default props if no props are passed", () => {
    props = loader.props();
    expect(props).toEqual(heightWidthRadiusDefaults(defaultHeight, defaultWidth, defaultRadius));
  });

  it("should contain styles created using default props", () => {
    for (let i = 0; i < 8; i++) {
      expect(loader.find("span span").at(i)).toHaveStyleRule("background-color", defaultColor);
      expect(loader.find("span span").at(i)).toHaveStyleRule(
        "height",
        `${defaultHeight}${defaultUnit}`
      );
      expect(loader.find("span span").at(i)).toHaveStyleRule(
        "width",
        `${defaultWidth}${defaultUnit}`
      );
      expect(loader.find("span span").at(i)).toHaveStyleRule(
        "margin",
        `${defaultMargin}${defaultUnit}`
      );
      expect(loader.find("span span").at(i)).toHaveStyleRule(
        "border-radius",
        `${defaultRadius}${defaultUnit}`
      );
    }
  });

  it("should render null if loading prop is set as false", () => {
    loader = mount(<FadeLoader loading={false} />);
    expect(loader.isEmptyRender()).toBe(true);
  });

  it("should render the correct color based on prop", () => {
    loader = mount(<FadeLoader color="#e2e2e2" />);
    expect(loader.find("span span")).not.toHaveStyleRule("background-color", defaultColor);
    expect(loader.find("span span")).toHaveStyleRule("background-color", "#e2e2e2");
  });

  describe("height props", () => {
    it("should render the height with px unit when height is a number", () => {
      const height = 18;
      loader = mount(<FadeLoader height={height} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "height",
          `${defaultHeight}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("height", `${height}${defaultUnit}`);
      }
    });

    it("should render the height as is when height is a string with valid css unit", () => {
      const height = "18px";
      loader = mount(<FadeLoader height={height} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "height",
          `${defaultHeight}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("height", `${height}`);
      }
    });

    it("should render the height with default unit of px when the unit is incorrect", () => {
      const length = 18;
      const unit = "ad";
      const height = `${length}${unit}`;
      loader = mount(<FadeLoader height={height} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "height",
          `${defaultHeight}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("height", `${length}${defaultUnit}`);
      }
    });
  });

  describe("width props", () => {
    it("should render the width with px unit when width is a number", () => {
      const width = 18;
      loader = mount(<FadeLoader width={width} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "width",
          `${defaultWidth}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("width", `${width}${defaultUnit}`);
      }
    });

    it("should render the width as is when width is a string with valid css unit", () => {
      const width = "18px";
      loader = mount(<FadeLoader width={width} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "width",
          `${defaultWidth}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("width", `${width}`);
      }
    });

    it("should render the width with default unit of px when the unit is incorrect", () => {
      const length = 18;
      const unit = "ad";
      const width = `${length}${unit}`;
      loader = mount(<FadeLoader width={width} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "width",
          `${defaultWidth}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("width", `${length}${defaultUnit}`);
      }
    });
  });

  describe("radius props", () => {
    it("should render the radius with px unit when radius is a number", () => {
      const radius = 18;
      loader = mount(<FadeLoader radius={radius} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "border-radius",
          `${defaultRadius}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule(
          "border-radius",
          `${radius}${defaultUnit}`
        );
      }
    });

    it("should render the radius as is when radius is a string with valid css unit", () => {
      const radius = "18px";
      loader = mount(<FadeLoader radius={radius} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "border-radius",
          `${defaultRadius}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("border-radius", `${radius}`);
      }
    });

    it("should render the radius with default unit of px when the unit is incorrect", () => {
      const length = 18;
      const unit = "ad";
      const radius = `${length}${unit}`;
      loader = mount(<FadeLoader radius={radius} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "border-radius",
          `${defaultRadius}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule(
          "border-radius",
          `${length}${defaultUnit}`
        );
      }
    });
  });

  describe("margin props", () => {
    it("should render the margin with px unit when margin is a number", () => {
      const margin = 18;
      loader = mount(<FadeLoader margin={margin} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "margin",
          `${defaultMargin}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("margin", `${margin}${defaultUnit}`);
      }
    });

    it("should render the margin as is when margin is a string with valid css unit", () => {
      const margin = "18px";
      loader = mount(<FadeLoader margin={margin} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "margin",
          `${defaultMargin}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("margin", `${margin}`);
      }
    });

    it("should render the margin with default unit of px when the unit is incorrect", () => {
      const length = 18;
      const unit = "ad";
      const margin = `${length}${unit}`;
      loader = mount(<FadeLoader margin={margin} />);

      for (let i = 0; i < 8; i++) {
        expect(loader.find("span span").at(i)).not.toHaveStyleRule(
          "margin",
          `${defaultMargin}${defaultUnit}`
        );

        expect(loader.find("span span").at(i)).toHaveStyleRule("margin", `${length}${defaultUnit}`);
      }
    });
  });

  it("should render the css override based on props", () => {
    loader = mount(
      <FadeLoader css={"position: absolute; width: 100px; height: 200px; color: blue;"} />
    );
    expect(loader).not.toHaveStyleRule("position", "relative");
    expect(loader).not.toHaveStyleRule("width", "60px");
    expect(loader).not.toHaveStyleRule("height", "60px");
    expect(loader).toHaveStyleRule("position", "absolute");
    expect(loader).toHaveStyleRule("width", "100px");
    expect(loader).toHaveStyleRule("height", "200px");
    expect(loader).toHaveStyleRule("color", "blue");
  });
});
