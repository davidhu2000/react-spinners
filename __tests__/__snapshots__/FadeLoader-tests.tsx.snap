// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeLoader should match snapshot 1`] = `
@keyframes animation-0 {
  50% {
    opacity: 0.3;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  50% {
    opacity: 0.3;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  50% {
    opacity: 0.3;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  50% {
    opacity: 0.3;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  50% {
    opacity: 0.3;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  50% {
    opacity: 0.3;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  50% {
    opacity: 0.3;
  }

  100% {
    opacity: 1;
  }
}

@keyframes animation-0 {
  50% {
    opacity: 0.3;
  }

  100% {
    opacity: 1;
  }
}

.emotion-8 {
  position: relative;
  font-size: 0;
  top: 20px;
  left: 20px;
  width: 60px;
  height: 60px;
}

.emotion-0 {
  position: absolute;
  width: 5px;
  height: 15px;
  margin: 2px;
  background-color: #000000;
  border-radius: 2px;
  -webkit-transition: 2s;
  transition: 2s;
  -webkit-animation-fill-mode: "both";
  animation-fill-mode: "both";
  -webkit-animation: animation-0 1.2s 0.12s infinite ease-in-out;
  animation: animation-0 1.2s 0.12s infinite ease-in-out;
  top: 20px;
  left: 0;
}

.emotion-1 {
  position: absolute;
  width: 5px;
  height: 15px;
  margin: 2px;
  background-color: #000000;
  border-radius: 2px;
  -webkit-transition: 2s;
  transition: 2s;
  -webkit-animation-fill-mode: "both";
  animation-fill-mode: "both";
  -webkit-animation: animation-0 1.2s 0.24s infinite ease-in-out;
  animation: animation-0 1.2s 0.24s infinite ease-in-out;
  top: 13.636363636363637px;
  left: 13.636363636363637px;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.emotion-2 {
  position: absolute;
  width: 5px;
  height: 15px;
  margin: 2px;
  background-color: #000000;
  border-radius: 2px;
  -webkit-transition: 2s;
  transition: 2s;
  -webkit-animation-fill-mode: "both";
  animation-fill-mode: "both";
  -webkit-animation: animation-0 1.2s 0.36s infinite ease-in-out;
  animation: animation-0 1.2s 0.36s infinite ease-in-out;
  top: 0;
  left: 20px;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-3 {
  position: absolute;
  width: 5px;
  height: 15px;
  margin: 2px;
  background-color: #000000;
  border-radius: 2px;
  -webkit-transition: 2s;
  transition: 2s;
  -webkit-animation-fill-mode: "both";
  animation-fill-mode: "both";
  -webkit-animation: animation-0 1.2s 0.48s infinite ease-in-out;
  animation: animation-0 1.2s 0.48s infinite ease-in-out;
  top: -13.636363636363637px;
  left: 13.636363636363637px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.emotion-4 {
  position: absolute;
  width: 5px;
  height: 15px;
  margin: 2px;
  background-color: #000000;
  border-radius: 2px;
  -webkit-transition: 2s;
  transition: 2s;
  -webkit-animation-fill-mode: "both";
  animation-fill-mode: "both";
  -webkit-animation: animation-0 1.2s 0.6s infinite ease-in-out;
  animation: animation-0 1.2s 0.6s infinite ease-in-out;
  top: -20px;
  left: 0;
}

.emotion-5 {
  position: absolute;
  width: 5px;
  height: 15px;
  margin: 2px;
  background-color: #000000;
  border-radius: 2px;
  -webkit-transition: 2s;
  transition: 2s;
  -webkit-animation-fill-mode: "both";
  animation-fill-mode: "both";
  -webkit-animation: animation-0 1.2s 0.72s infinite ease-in-out;
  animation: animation-0 1.2s 0.72s infinite ease-in-out;
  top: -13.636363636363637px;
  left: -13.636363636363637px;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.emotion-6 {
  position: absolute;
  width: 5px;
  height: 15px;
  margin: 2px;
  background-color: #000000;
  border-radius: 2px;
  -webkit-transition: 2s;
  transition: 2s;
  -webkit-animation-fill-mode: "both";
  animation-fill-mode: "both";
  -webkit-animation: animation-0 1.2s 0.84s infinite ease-in-out;
  animation: animation-0 1.2s 0.84s infinite ease-in-out;
  top: 0;
  left: -20px;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-7 {
  position: absolute;
  width: 5px;
  height: 15px;
  margin: 2px;
  background-color: #000000;
  border-radius: 2px;
  -webkit-transition: 2s;
  transition: 2s;
  -webkit-animation-fill-mode: "both";
  animation-fill-mode: "both";
  -webkit-animation: animation-0 1.2s 0.96s infinite ease-in-out;
  animation: animation-0 1.2s 0.96s infinite ease-in-out;
  top: 13.636363636363637px;
  left: -13.636363636363637px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<Loader
  color="#000000"
  css=""
  height={15}
  loading={true}
  margin={2}
  radius={2}
  width={5}
>
  <span
    className="emotion-8"
  >
    <span
      className="emotion-0"
    />
    <span
      className="emotion-1"
    />
    <span
      className="emotion-2"
    />
    <span
      className="emotion-3"
    />
    <span
      className="emotion-4"
    />
    <span
      className="emotion-5"
    />
    <span
      className="emotion-6"
    />
    <span
      className="emotion-7"
    />
  </span>
</Loader>
`;
